lib/elsa/producer.ex

```diff
@@ -23,7 +23,7 @@ defmodule Elsa.Producer do
   @spec produce(
           Elsa.endpoints() | Elsa.connection(),
           Elsa.topic(),
-          {term(), term()} | term() | [{term(), term()}] | [term()],
+          {iodata(), iodata()} | binary() | [{iodata(), iodata()}] | [iodata()],
```

Comment:
```suggestion
          {iodata(), iodata()} | iodata() | [{iodata(), iodata()}] | [iodata()],
```

Since `iodata` == `iolist | binary`, you can have a single `iodata` input here, correct?

Reply:
In this case, no since I can produce a single binary, but a single iolist will be parsed as a list of binaries or a list tuples and then blow up when it doesn’t fit that pattern. To produce a single-message iolist you’d need to use the last one and wrap that message in one extra list layer

Reply:
Although I now realize that turning `term()` to `binary()` means I’ve lost the spec for a single map as a message which we do allow, so I need to update this and probably break the allowable message out into its own type doc